### PR TITLE
Metadata for Tup build systems

### PR DESCRIPTION
**Pull request by ghost**
This pull request is for adding some minimal metadata for [Tup](http://gittup.org/tup) build systems. Each core gets a `Tupfile` that loads rules from a parent directory above this repository.

I am moving the Libretro build system for the [Genode](http://genode.org) platform from Make to a Tup super-repo using Git submodules. By adding this metadata "upstream" I only build these cores if the core submodule is checked out. I build all the cores at once so that I can tweak things like the linker and see the effects on every core. As an example my build rules that sit above this submodule are here: https://github.com/ehmry/genode-emery/blob/tup/libretro/cores/samples/Tuprules.tup.

If this is okay, I will start adding tupfiles to the other core repositories as I port them to Genode. To keep this nice and tidy I don't want to duplicate any information with the makefiles other than `TARGET_NAME` and `CORE_DIR`, my intention is to move messy platform rules out of the core repositories.

### CODE REVIEW

**Comment by ghost** [comment]
Also, some minor patches to make testing cores easier for me.

**Comment by RobLoach** [comment]
Why Tup? Looks the same as a `Makefile`...

Some questions...

- Are we duplicating effort?
- Can you install `make` on Genode?
- What rules out other widely adopted build systems like [CMake](https://cmake.org)?

File: midi/midi_test/libretro.c [hunk 1]
```diff
@@ -190,7 +190,7 @@ void retro_get_system_av_info(struct retro_system_av_info *info)
    info->geometry.max_height   = HEIGHT;
    info->geometry.max_width    = WIDHT;
    info->timing.fps            = FPS;
-   info->timing.sample_rate    = 48000.0;
```
Comment (RobLoach) [inline midi/midi_test/libretro.c:193]: I believe the midi test may need a sample rate.

Comment (ghost) [reply midi/midi_test/libretro.c:193]: The MIDI test discards the sampled audio callbacks from the frontend, so I assumed there would be no need for a sample rate.

**Comment by ghost** [comment]
@RobLoach Tup is different from Make and CMake in that the build metadata is at the bottom of the tree rather than the top, so the build system gets rooted somewhere and searches its subdirectories for Tupfiles, and builds its recipes from the sum of these. This allows it to build a bunch of smaller git repositories as one big job, which is useful to me because I'm building the Genode frontend, all the ported cores, and the packaging metadata at once.

Some of the cores are already ported to Genode and have Make rules, but as I see it this is mostly duplicated effort because I have to commit the Genode build rules in each core repository, and I'm not confident that I wouldn't want to changes these rules in the future. This also encourages a clean separation of generic and platfrom build info between `Makefile.libretro` and `Makefile.common` because Tup can parse parse simple `Makefile.common` files.

Also, this saves me and maybe others from having to integrate our exotic platforms into libretro-super.

**Pull request merged**